registry/registry.go

```diff
@@ -87,27 +87,15 @@ func (r *RouteRegistry) Register(uri route.Uri, endpoint *route.Endpoint) {
 		return
 	}
 
-	endpointAdded := r.register(uri, endpoint)
+	endpointPutResult := r.register(uri, endpoint)
 
-	r.reporter.CaptureRegistryMessage(endpoint)
-
-	if endpointAdded == route.ADDED && !endpoint.UpdatedAt.IsZero() {
+	if endpointPutResult == route.EndpointAdded && !endpoint.UpdatedAt.IsZero() {
 		r.reporter.CaptureRouteRegistrationLatency(time.Since(endpoint.UpdatedAt))
 	}
 
-	switch endpointAdded {
-	case route.ADDED:
-		if r.logger.Enabled(context.Background(), slog.LevelInfo) {
-			r.logger.Info("endpoint-registered", buildSlogAttrs(uri, endpoint)...)
-		}
-	case route.UPDATED:
-		if r.logger.Enabled(context.Background(), slog.LevelDebug) {
-			r.logger.Debug("endpoint-registered", buildSlogAttrs(uri, endpoint)...)
-		}
-	default:
-		if r.logger.Enabled(context.Background(), slog.LevelDebug) {
-			r.logger.Debug("endpoint-not-registered", buildSlogAttrs(uri, endpoint)...)
-		}
+	r.reporter.CaptureRegistryMessage(endpoint, string(endpointPutResult))
+	if r.logger.Enabled(context.Background(), slog.LevelInfo) {
+		r.logger.Info(string(endpointPutResult), buildSlogAttrs(uri, endpoint)...)
```

Comment:
this will significantly change the log output, as we'll have log messages for each updated (including no-op) and 'not-registered' messages, right?

Reply:
We changed it. Only changes will be logged with info level.